glue/core/hub.py

```diff
@@ -175,9 +182,12 @@ def _find_handlers(self, message):
             # narrow to the most-specific message
             candidate = max(messages, key=_mro_count)
 
-            handler, test = subscriptions[candidate]
+            handler, test, priority = subscriptions[candidate]
             if test(message):
-                yield subscriber, handler
+                prioritized_handlers.append((subscriber, handler, priority))
+
+        for subscriber, handler, _ in sorted(prioritized_handlers, key=lambda x: x[2], reverse=True):
```

Comment:
Should this loop be moved up one level in indentation?

Reply:
I pushed that fix as you were typing! That was the problem.

Reply:
Need to remember to force-reload GH interface before looking into stuff!